gui/src/app/Stanc/Types.ts

```diff
@@ -0,0 +1,33 @@
+export type StancErrors = {
+  errors?: string[];
+  warnings?: string[];
+};
+
+type StancReturn = { result?: string } & StancErrors;
+
+export type StancFunction = (
+  name: string,
+  code: string,
+  args: string[],
+) => StancReturn;
+
+export enum Replies {
+  Formatted = "formatted",
+  Checked = "checked",
+}
+
+export type StancReplyMessage = {
+  purpose?: Replies;
+  error: string;
+} & StancReturn;
+
+export enum Requests {
+  Format = "format",
+  Check = "check",
+}
+
+export type StancRequestMessage = {
+  purpose: Requests;
+  name: string;
+  code: string;
+};
```

Comment:
At the risk of being annoying--would you be willing to be a bit more specific with the names of the `Requests` and `Replies` enums? Like `StancWorkerRequests`?

Additionally, the `Format` and `Check` members could be more descriptive, when they appear elsewhere, as `SyntaxCheck` or `ReformatStanCode` or something like this. (I'm not attached to the specifics, just noticing that `Replies.Formatted` in `useStanc.ts` reads casually as looking at replies that are formatted, rather than the Reply that says "hey I formatted your code")

Reply:
This and your above comment made me really think hard about what actually is needed, and I realized that there was some redundancy, so I cleaned a bunch up. Take another look?